### PR TITLE
Add options for client to node encryption (SSL)

### PR DESCRIPTION
**Pull request by mpenick**
Fixes: https://github.com/datastax/cassandra-data-apis/issues/58

### CODE REVIEW

**Comment by mpenick** [comment]
Tested manually on a local cluster. Working on an automated test as part of: https://github.com/datastax/cassandra-data-apis/issues/52

File: cmd/server.go [hunk 1]
```diff
@@ -165,9 +178,26 @@ func createEndpoint() *endpoint.DataEndpoint {
 		updateInterval = endpoint.DefaultSchemaUpdateDuration
 	}
 
+	var sslOptions *db.SslOptions
+
+	sslCaCertPath := viper.GetString("ssl-ca-cert-path")
+	sslCertPath := viper.GetString("ssl-client-cert-path")
+	sslKeyPath := viper.GetString("ssl-client-key-path")
+	if sslCaCertPath != "" || (sslCertPath != "" && sslKeyPath != "") {
```
Comment (jorgebay) [inline cmd/server.go:186]: I think we should also allow enabling TLS without a client cert o a CA.

File: db/db.go [hunk 2]
```diff
@@ -25,8 +25,21 @@ type Db struct {
...
 }
 
+type SslOptions struct {
+	CaPath string
+	CertPath string
+	KeyPath string
+	HostVerification bool
+}
+
+type Config struct {
```
Comment (jorgebay) [inline db/db.go:35]: 💯 a dedicated db config struct.

File: README.md [hunk 3]
```diff
@@ -131,6 +136,27 @@ JSON:
...
+By default, traffic between the API endpoints and the database servers is not encrypted. To secure
+this traffic you will need to generate SSL certificates and enable SSL on the database servers. More
+information about enabling SSL (client-to-node encryption) on the database servers can be found in
+this [documentation][client-to-node]. After SSL is enabled on the database servers use the
+`ssl-enabled` option, along with `ssl-ca-cert-path` to enable secure connections. `ssl-ca-cert-path`
+is a path to the chain of certificates used to generate the database server's certificates. The
+certificate chain is used by API endpoints to verify the database server's certificates.  The
+`ssl-client-cert-path` and `ssl-client-key-path` options are not required, but can be use to provide
+client-side certificates that are used by the database servers to authenticate and verify the API
+servers, this is know as mutual authentication.
```
Comment (jorgebay) [inline README.md:158]: Small typo.

```suggestion
servers, this is known as mutual authentication. 
```

**Pull request merged**